BUILDING.md

```diff
@@ -0,0 +1,249 @@
+
+```
+$ sudo apt-get update
+$ sudo apt-get install -y build-essential cmake git python3-pip
+$ sudo pip3 install conan
+$ sudo apt-get install libgtk2.0-dev libasound2-dev libavformat-dev libjack-jackd2-dev
+```
+
+## Building on Windows
+
+1. Clone Audacity from the Audacity GitHub project. 
+  
+   For example, in the **git-bash** run:
+
+    ```
+    $ git clone https://github.com/audacity/audacity/
+    ```
+
+2. Open CMake GUI. 
+   
+   Set the **Where is the source code** to the location Audicity was cloned. 
+   
+   Set **Where to build the binaries** to the location you want to place your build in. It is preffered, that this location is not within the directory with the source code.
+
+3. Press **Configure**. You can choose which version of Visual Studio to use and the platform to build for in the popup. We support **x64** and **Win32** platforms. The **x64** platform is a default option. Press **Finish** to start the configuration process.
+
+4. After the successful configuration, you will see `Configuring done` in the last line of the log. Press **Generate** to generate the Visual Studio project. 
+
+5. After you see "Generating done", press **Open Project** to open the project in Visual Studio.
+   
+6. Select "Build -> Build Solution".
+   
+7. You can now run and debug Audacity!
+      
+Generally, steps 1-5 are only needed for the first-time configuration. Then, after you've generated the solution - you can open it in Visual Studio next time. In case when the project configuration has changed - the IDE will invoke CMake internally. 
+
+> Conan Center provides prebuilt binaries only for **x64**. Configuring the project for Win32 will take much longer, as all the 3d party libraries will be built during the configuration.
+
+## macOS
+
+1. Clone Audacity from the Audacity GitHub project. 
+  
+    ```
+    $ git clone https://github.com/audacity/audacity/
+    ```
+
+2. Configure Audacity using CMake:
+   ```
+   $ mkdir build && cd build
+   $ cmake -GXCode -T buildsystem=1 ../audacity
+   ```
+
+3. Open Audacity XCode project:
+   ```
+   $ open Audacity.xcodeproj
+   ```
+   and build Audacity using the IDE. 
+
+Steps 1 and 2 are only required for the first-time builds. 
+
+Alternatively, you can use **CLion**. If you chose so, open the directory where you have cloned Audacity using CLion and you are good to go.
+
+At the moment, we only support **x86_64** builds. It is possible to build using AppleSilicon harware, but **mad** and **id3tag** should be disabled:
+
+```
+cmake -GXCode -T buildsystem=1 -Daudacity_use_mad="off" -Daudacity_use_id3tag=off ../audacity
+```
+
+## Linux & Other OS
+
+1. Clone Audacity from the Audacity GitHub project. 
+  
+    ```
+    $ git clone https://github.com/audacity/audacity/
+    ```
+
+2. Configure Audacity using CMake:
+   ```
+   $ mkdir build && cd build
+   $ cmake -G "Unix Makefiles" -Daudacity_use_ffmpeg=loaded ../audacity
```

Comment:
```suggestion
   $ cmake -Daudacity_use_ffmpeg=loaded ../audacity
```
Unix Makefiles is the default generator on Linux and macOS. There is no need to specify it explicitly (I personally use Ninja though).

Reply:
I'd rather prefer to have it explicit